### ISSUE TITLE
No Git history for selected file

### ISSUE THREAD

**Issue by willyyang** [jupyterlab/jupyterlab-git#1024]
## Description

Unable to view Git history for a selected file. Implemented in https://github.com/jupyterlab/jupyterlab-git/commit/02c6d9aa8cf3344b37397e303c8d345efab12aff for  https://github.com/jupyterlab/jupyterlab-git/issues/864

## Reproduce

1. Clone an active repo 
2. Right click on a file (containing history) and navigate to `Git` -> `History`
3. `No history found` shows on file

## Expected behavior

Expect to view a number of Git history commits based on the value set for `historyCount` under advanced settings (kept default of 25) 

## Context

```
# cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux 10 (buster)"

# pip list | grep jupyterlab-git
jupyterlab-git         0.32.2

# jupyter labextension list
JupyterLab v3.0.17
  @jupyterlab/git v0.32.2 enabled OK (python, jupyterlab-git)

# jupyter server extension list
  jupyterlab_git enabled
  - Validating jupyterlab_git...
     jupyterlab_git 0.32.2 OK

# git --version
git version 2.20.1
```

**Comment by adammclaurin** [comment]
@navn-r fyi

**Comment by navn-r** [comment]
Thanks @willyyang for pointing out the issue and thanks @adammclaurin for pinging me on it.

The bug is definitely reproducible on my side. One thing to note that is the file history works when right clicking on the file in the Git Panel. But you'd need to have modified the file in order to do so, so I can't say to do that as a fix/workaround.

I've found that doing something like this internally should do the trick. 

```bash
git -C cloned-repo-folder log ...
```
`-C` is the flag to specify the exact git folder, [source](https://stackoverflow.com/questions/3769137/use-git-log-command-in-another-folder#3769169)


I'd be more than glad to get a PR in for it. @fcollonval is there any other approach we could take? Considering the regular repository history works as intended.

**Comment by fcollonval** [comment]
Hey @willyyang thanks for reporting the bug. Could you confirm the workflow leading to the bug?

1. Clone a repository using the Git menu or Git clone button of the filebrowser
2. Navigate inside the cloned repository with the filebrowser
3. Right-click on a file and select the context menu entry `Git -> History`
4. The Git history panel appears but it displays an error instead of the expected file history

Are you using the standard file browser or a custom extension?

**Comment by willyyang** [comment]
Hey @fcollonval , I'm using the standard file browser . 
For #4, there is no error displayed just `No history found`.

**Comment by fcollonval** [comment]
> Hey @fcollonval , I'm using the standard file browser . 
> For 4, there is no error displayed just `No history found`.


Thanks for the quick answer. I pushed a fix.

**Comment by willyyang** [comment]
Thanks for the quick fix @fcollonval . I see a new release tag has been cut, when will this change be published?

**Comment by fcollonval** [comment]
Initially I wanted to do a release right away. But the new merge conflict view (that is part of master) is not ready yet for publication (remaining tasks are tracked in https://github.com/jupyterlab/jupyterlab-git/pull/1031).

But I may branch out to cut a patch release for this one.

**Comment by fcollonval** [comment]
@willyyang I released 0.32.3 with the fix.
